translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java

```diff
@@ -0,0 +1,363 @@
+            stepData.put("Translator", translator);
+        } catch (Exception ex) {
+            verifyException(ex);
+        }
+    }
+
+    @Then("^Translator \"([^\"]*)\" is found$")
+    public void translatorIsFound(String translatorName) {
+        Translator translator = (Translator) stepData.get("Translator");
+        assertEquals(translatorName, translator.getClass().getSimpleName());
+    }
+
+    @Given("^I create mqtt message with (?:valid|invalid|empty) payload \"([^\"]*)\" and (?:valid|invalid) topic \"([^\"]*)\"$")
+    public void iTryToProvideInvalidTopic(String payload, String topic) {
+
+        Date date = new Date();
+        MqttTopic mqttTopic = new MqttTopic(topic);
+
+        KuraPayload kuraPayload = new KuraPayload();
+        if (payload.equals("invalidPayload") || payload.equals("")) {
+            kuraPayload.setBody(payload.getBytes());
+        } else {
+            kuraPayload.getMetrics().put(payload, 200);
+        }
+
+        MqttPayload mqttPayload = new MqttPayload(kuraPayload.toByteArray());
+        MqttMessage mqttMessage = new MqttMessage(mqttTopic, date, mqttPayload);
+
+        stepData.put("MqttMessage", mqttMessage);
+    }
+
+    @When("^I try to translate mqtt response$")
+    public void iTryToTranslateMqttResponse() throws Exception {
+        MqttMessage mqttMessage = (MqttMessage) stepData.get("MqttMessage");
+
+        try {
+            KuraResponseMessage kuraResponseMessage = translatorResponseMqttKura.translate(mqttMessage);
+            stepData.put("KuraResponseMessage", kuraResponseMessage);
+        } catch (KapuaException ex) {
+            verifyException(ex);
+        }
+    }
+
+    @Then("^I got kura response message with \"([^\"]*)\" payload body$")
+    public void iGotKuraResponseMessageWithPayloadBody(String payloadType) {
+        KuraResponseMessage kuraResponseMessage = (KuraResponseMessage) stepData.get("KuraResponseMessage");
+
+        assertTrue(kuraResponseMessage.getPayload().getBody().getClass().getSimpleName().equals(payloadType));
+    }
+
+    @Then("^I got kura response message with proper payload metrics$")
+    public void iGotKuraResponseMessageWithPayloadAndChannelWithAndData() {
+        KuraResponseMessage kuraResponseMessage = (KuraResponseMessage) stepData.get("KuraResponseMessage");
+
+        assertTrue(kuraResponseMessage.getPayload().getMetrics() != null);
+    }
+
+    @Given("^I create kura data message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and payload without body and metrics$")
+    public void iCreateKuraDataMessage(String scope, String clientId) {
+        KuraDataChannel kuraDataChannel = new KuraDataChannel(scope, clientId);
+        Date date = new Date();
+        KuraDataPayload kuraDataPayload = new KuraDataPayload();
+        KuraDataMessage kuraDataMessage = new KuraDataMessage(kuraDataChannel, date, kuraDataPayload);
```

Comment:
insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.